pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -736,6 +790,35 @@ func CheckNodeAgentLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.L
 	return nil
 }
 
+// CheckForExpiredNodeLeases checks if the number of expired node Leases surpasses 20% of all existing Leases. If yes,
+// an error will be returned. The motivation is that dependency-watchdog is starting to scale down controllers when 60%
+// of the Leases are expired.
+func CheckForExpiredNodeLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.LeaseList, clock clock.Clock) error {
+	if len(leaseList.Items) == 0 || len(nodeList.Items) == 0 {
+		return nil
+	}
+
+	nodeNames := sets.Set[string]{}
+	for _, node := range nodeList.Items {
+		nodeNames.Insert(node.Name)
+	}
+
+	var expiredLeases int
+	for _, lease := range leaseList.Items {
+		if lease.Spec.RenewTime.Add(time.Second*time.Duration(*lease.Spec.LeaseDurationSeconds)).Before(clock.Now()) &&
+			// we only care about Leases related to existing nodes
+			nodeNames.Has(lease.Name) {
```

Comment:
```suggestion
                // we only care about Leases related to existing nodes
                nodeNames.Has(lease.Name)  &&
		if lease.Spec.RenewTime.Add(time.Second*time.Duration(*lease.Spec.LeaseDurationSeconds)).Before(clock.Now()) &&
```

I saw that `LeaseDurationSeconds` and `RenewTime` are an optional fields. I think we should check whether it is a node lease first before the expiry check as it is possible that some leases do not have these fields set.